x-pack/solutions/observability/plugins/observability/server/routes/alerts/route.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  getRelatedDashboardsParamsSchema,
+  GetRelatedDashboardsResponse,
+} from '@kbn/observability-schema';
+import { IKibanaResponse } from '@kbn/core-http-server';
+import type { SavedObjectsFindResult } from '@kbn/core/server';
+import type { DashboardAttributes } from '@kbn/dashboard-plugin/server/content_management/v3';
+import { createObservabilityServerRoute } from '../create_observability_server_route';
+import { SuggestedDashboardsClient } from '../../services/suggested_dashboards_client';
+import { InvestigateAlertsClient } from '../../services/investigate_alerts_client';
+import { AlertNotFoundError } from '../../common/errors/alert_not_found_error';
+
+const alertsDynamicDashboardSuggestions = createObservabilityServerRoute({
+  endpoint: 'GET /internal/observability/alerts/suggested_dashboards',
+  security: {
+    authz: {
+      enabled: false,
+      reason:
+        'This route is opted out from authorization because it is a wrapper around Saved Object client',
+    },
+  },
+  options: { access: 'internal' },
+  params: getRelatedDashboardsParamsSchema,
+  handler: async (services): Promise<GetRelatedDashboardsResponse | IKibanaResponse> => {
+    const { dependencies, params, request, response, context, logger } = services;
+    const { alertId } = params.query;
+    const { ruleRegistry, dashboard } = dependencies;
+    const { contentClient } = dashboard;
+    const dashboardClient = contentClient!.getForRequest<
+      SavedObjectsFindResult<DashboardAttributes>
+    >({
+      requestHandlerContext: context,
+      request,
+      version: 3,
+    });
+
+    const alertsClient = await ruleRegistry.getRacClientWithRequest(request);
+    const investigateAlertsClient = new InvestigateAlertsClient(alertsClient);
+
+    const dashboardParser = new SuggestedDashboardsClient(
+      logger,
+      dashboardClient,
+      investigateAlertsClient,
+      alertId
+    );
+    try {
+      const { suggestedDashboards } = await dashboardParser.fetchSuggestedDashboards();
+      return {
+        suggestedDashboards,
+        linkedDashboards: [],
```

Comment:
@dominiqueclarke The idea is to eventually use something like this `await dashboardParser.fetchLinkedDashboards();` in order to fetch the linkedDashboards? I would like us to sync up regarding this. In my PR I created a [dashboard service](https://github.com/elastic/kibana/pull/209668/files#diff-c028c09ef64d63b0fb8976dfdc836d9bf50836039279ba18a635a000ae83fbf8) to simply get all dashboards and I want to discuss with you how we plan to move forward

Reply:
Yup! We discussed this offline and arrive to the same page.